commit 071bba328d62a0df87d0b4d94797681c3850ac9a
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Wed Aug 16 16:52:50 2017 +0100

    BCM270X_DT: Set spidev spi-max-frequency to 125MHz
    
    The BCM2835 SPI controllers have a maximum bus clock of half the system
    clock speed, so with the 250MHz system clock found on Raspberry Pis
    you get a theoretical maximum bus speed of 125MHz. Note that this
    speed is unlikely to be reliable, and the maximum usable bus spee will
    depend on both the attached device and the wiring.
    
    See: https://github.com/raspberrypi/linux/issues/2165
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/boot/dts/bcm2708-rpi-0-w.dts b/arch/arm/boot/dts/bcm2708-rpi-0-w.dts
index db0f99ddf2f4..d585f052d580 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-0-w.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-0-w.dts
@@ -108,7 +108,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -116,7 +116,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
index 08bf838fab55..ecbcfc35f2a8 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
@@ -69,7 +69,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -77,7 +77,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/bcm2708-rpi-b.dts b/arch/arm/boot/dts/bcm2708-rpi-b.dts
index 4e6b4dd6a8d9..2407c23d1eea 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-b.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b.dts
@@ -69,7 +69,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -77,7 +77,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/bcm2708-rpi-cm.dts b/arch/arm/boot/dts/bcm2708-rpi-cm.dts
index 10be69972bd1..742cb65dedc1 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-cm.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-cm.dts
@@ -52,7 +52,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -60,7 +60,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/bcm2709-rpi-2-b.dts b/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
index 2dc0e1204e63..7aea36283cbb 100644
--- a/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
+++ b/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
@@ -69,7 +69,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -77,7 +77,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
index cbec919208f3..aadd762fae13 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -128,7 +128,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -136,7 +136,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
index fe402e84cdda..b8b5cbc1b24c 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
@@ -89,7 +89,7 @@
 		reg = <0>;	/* CE0 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 
 	spidev1: spidev@1{
@@ -97,7 +97,7 @@
 		reg = <1>;	/* CE1 */
 		#address-cells = <1>;
 		#size-cells = <0>;
-		spi-max-frequency = <500000>;
+		spi-max-frequency = <125000000>;
 	};
 };
 
diff --git a/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts b/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts
index 71c243947004..c3d4f96b7aa9 100644
--- a/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi1-1cs-overlay.dts
@@ -36,7 +36,7 @@
 				reg = <0>;      /* CE0 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm/boot/dts/overlays/spi1-2cs-overlay.dts b/arch/arm/boot/dts/overlays/spi1-2cs-overlay.dts
index 2ae0885cf1e7..2ad62497dc89 100644
--- a/arch/arm/boot/dts/overlays/spi1-2cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi1-2cs-overlay.dts
@@ -36,7 +36,7 @@
 				reg = <0>;      /* CE0 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 
@@ -45,7 +45,7 @@
 				reg = <1>;      /* CE1 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm/boot/dts/overlays/spi1-3cs-overlay.dts b/arch/arm/boot/dts/overlays/spi1-3cs-overlay.dts
index 8f79044cd8f4..ef82890453bf 100644
--- a/arch/arm/boot/dts/overlays/spi1-3cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi1-3cs-overlay.dts
@@ -36,7 +36,7 @@
 				reg = <0>;      /* CE0 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 
@@ -45,7 +45,7 @@
 				reg = <1>;      /* CE1 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 
@@ -54,7 +54,7 @@
 				reg = <2>;      /* CE2 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm/boot/dts/overlays/spi2-1cs-overlay.dts b/arch/arm/boot/dts/overlays/spi2-1cs-overlay.dts
index 6f57bc710fe1..761b6be4ff9b 100644
--- a/arch/arm/boot/dts/overlays/spi2-1cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi2-1cs-overlay.dts
@@ -36,7 +36,7 @@
 				reg = <0>;      /* CE0 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm/boot/dts/overlays/spi2-2cs-overlay.dts b/arch/arm/boot/dts/overlays/spi2-2cs-overlay.dts
index d090631d3682..e533aba113de 100644
--- a/arch/arm/boot/dts/overlays/spi2-2cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi2-2cs-overlay.dts
@@ -36,7 +36,7 @@
 				reg = <0>;      /* CE0 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 
@@ -45,7 +45,7 @@
 				reg = <1>;      /* CE1 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 		};
diff --git a/arch/arm/boot/dts/overlays/spi2-3cs-overlay.dts b/arch/arm/boot/dts/overlays/spi2-3cs-overlay.dts
index e2586728c311..a62e107dc98f 100644
--- a/arch/arm/boot/dts/overlays/spi2-3cs-overlay.dts
+++ b/arch/arm/boot/dts/overlays/spi2-3cs-overlay.dts
@@ -36,7 +36,7 @@
 				reg = <0>;      /* CE0 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 
@@ -45,7 +45,7 @@
 				reg = <1>;      /* CE1 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 
@@ -54,7 +54,7 @@
 				reg = <2>;      /* CE2 */
 				#address-cells = <1>;
 				#size-cells = <0>;
-				spi-max-frequency = <500000>;
+				spi-max-frequency = <125000000>;
 				status = "okay";
 			};
 		};
